src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs

```diff
@@ -572,7 +574,8 @@ public static bool TrySkipBlock(Stream stream)
             // Already read the signature, so make the filename length field location relative to that
             stream.Seek(FieldLocations.FilenameLength - FieldLengths.Signature, SeekOrigin.Current);
 
-            bytesRead = stream.Read(blockBytes);
+            Debug.Assert(blockBytes.Length == FieldLengths.FilenameLength + FieldLengths.ExtraFieldLength);
```

Comment:
I'm not sure I understand this debug. Wasn't blockBytes.Length fixed to 4 bytes? The debug above (which I suggested to remove) was checking a different length, but I don't see that blockBytes is getting replaced with an array of different size.

Reply:
I've added them because `blockBytes` is used twice: once to store the four signature bytes, and once to hold the filename length & extra field length (which also both fill four bytes.) The asserts were to make it clearer that this was happening, but I've added a comment too. I can remove the assertions and just use the comment if you prefer